application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java

```diff
@@ -78,14 +82,39 @@ public <E extends HasName, I extends EntityId> void notifyDeleteEntity(TenantId
     }
 
     public void notifyDeleteAlarm(TenantId tenantId, Alarm alarm, EntityId originatorId,
-                                  CustomerId customerId, ActionType actionType,
+                                  CustomerId customerId,
                                   List<EdgeId> relatedEdgeIds,
                                   SecurityUser user,
                                   String body, Object... additionalInfo) {
-        logEntityAction(tenantId, originatorId, alarm, customerId, actionType, user, additionalInfo);
+        logEntityAction(tenantId, originatorId, alarm, customerId, ActionType.DELETED, user, additionalInfo);
         sendAlarmDeleteNotificationMsg(tenantId, alarm, relatedEdgeIds, body);
     }
 
+    @Override
+    public void notifyDeleteRuleChane(TenantId tenantId, RuleChain ruleChain,
+                                      List<EdgeId> relatedEdgeIds, SecurityUser user) {
+        RuleChainId ruleChainId = ruleChain.getId();
+        logEntityAction(tenantId, ruleChainId, ruleChain, null, ActionType.DELETED, user, ruleChainId.toString());
+        if (RuleChainType.EDGE.equals(ruleChain.getType())) {
+            sendDeleteNotificationMsg(tenantId, ruleChainId, relatedEdgeIds, null);
+        }
+    }
+
+    @Override
+    public void notifySaveRuleChainMetaData(TenantId tenantId, RuleChain ruleChain, Object... others) {
```

Comment:
This method looks more complicated from the original version.
Please pass variables as it is
`void notifySaveRuleChainMetaData(TenantId tenantId, RuleChain ruleChain, RuleChain updatedRuleChain, RuleChainMetaData updatedRuleChainMetaData, SecurityUser user);`
and create one more method only for single rule chain
`void notifySaveRuleChainMetaData(TenantId tenantId, RuleChain ruleChain);`
that is going to be used to identify edge.
This is much clearer and more readable.

Reply:
or please use for edge previous approach:
`void notifySaveRuleChainMetaData(TenantId tenantId, RuleChainId ruleChainId, Object... others);`
and then in the service:
```
            if (RuleChainType.EDGE.equals(ruleChain.getType())) {
                notificationEntityService.notifySaveRuleChainMetaData(tenantId, ruleChain);
                updatedRuleChains.forEach(updatedRuleChain -> {
                    notificationEntityService.notifySaveRuleChainMetaData(updatedRuleChain.getTenantId(), updatedRuleChain.getId());
                });
            }
```

Reply:
Ok